fs/btrfs/zstd.c

```diff
@@ -53,7 +69,7 @@ static void zstd_free_workspace(struct list_head *ws)
 static struct list_head *zstd_alloc_workspace(void)
 {
 	ZSTD_parameters params =
-			zstd_get_btrfs_parameters(ZSTD_BTRFS_MAX_INPUT);
+			zstd_get_btrfs_parameters(ZSTD_BTRFS_MAX_INPUT, ZSTD_BTRFS_DEFAULT_LEVEL);
```

Comment:
Can the initial allocation use the configured level instead of requiring a reallocation the first time that it's used? Then, the only time we need to reallocate is if the user remounts with a different level, which should be pretty rare.